pydrive2/auth.py

```diff
@@ -388,16 +413,20 @@ def SaveCredentialsFile(self, credentials_file=None):
         """
         if self.credentials is None:
             raise InvalidCredentialsError("No credentials to save")
+
         if credentials_file is None:
-            credentials_file = self.settings.get("save_credentials_file")
-            if credentials_file is None:
+            storage = self._storages["file"]
+            if storage is None:
                 raise InvalidConfigError(
-                    "Please specify credentials file to read"
+                    "Backend `file` is not configured, specify "
+                    "credentials file to read in the settings "
+                    "file or pass an explicit value"
                 )
-        try:
+        else:
             storage = Storage(credentials_file)
+
+        try:
             storage.put(self.credentials)
-            self.credentials.set_store(storage)
```

Comment:
For the record - removing this line here and moving `storage = Storage(credentials_file)` is the most important part of this change.

This way storage initialized once instead of every time we are refreshing the token.